src/wp-includes/option.php

```diff
@@ -965,7 +968,11 @@ function add_option( $option, $value = '', $deprecated = '', $autoload = 'yes' )
 	}
 
 	$serialized_value = maybe_serialize( $value );
-	$autoload         = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+
+	if ( null === $autoload ) {
+		$autoload = sanitize_option_size( $option, $serialized_value );
+	}
+	$autoload = sanitize_autoload( $autoload );
```

Comment:
Let's put this into an `else`. No need to sanitize the value if it comes from WP core's own function that already returns only safe values.
```suggestion
	} else {
		$autoload = sanitize_autoload( $autoload );
	}
```